### ISSUE TITLE
const error

### ISSUE THREAD

**Issue by skaller** [joyent/libuv#1036]
```
  CC       src/unix/libuv_la-fs.lo
src/unix/fs.c:216:34: warning: incompatible pointer types passing 'int (const struct dirent *)' to parameter of
      type 'int (*)(struct dirent *)' [-Wincompatible-pointer-types]
  n = scandir(req->path, &dents, uv__fs_readdir_filter, alphasort);
                                 ^~~~~~~~~~~~~~~~~~~~~
/usr/include/dirent.h:129:11: note: passing argument to parameter here
    int (*)(struct dirent *), int (*)(const void *, const void *)) __DARWIN_INODE64(scandir);
          ^
1 warning generated.
```

Detected by clang.

**Comment by indutny** [comment]
Hello!

Thanks for opening an issue, what OSX version are you using? `scandir()` has a filter function with `const` argument on all platforms available to me: OSX 10.9, ubuntu, smartos.

**Comment by jneu** [comment]
Hi @skaller - would you mind compiling jneu/libuv@3fb800b131ab6ff23e16110e88f3d494ef8bf0c6 to see if this fixes the issue? I don't have access to OS X 10.7 to verify :-)

**Comment by saghul** [comment]
Fixed by https://github.com/joyent/libuv/commit/daa457b1c043b46fb0936bb08787752f1daa445e.
